signac/contrib/project.py

```diff
@@ -156,6 +156,16 @@ def __init__(self, config=None, _ignore_schema_version=False):
         self._fn_doc = os.path.join(self._rd, self.FN_DOCUMENT)
         self._document = None
 
+        # Prepare Workspace Directory
+        if not os.path.isdir(self._wd):
+            try:
+                _mkdir_p(self._wd)
+            except OSError:
+                logger.error(
+                    "Error occurred while trying to create "
+                    "workspace directory for project {}.".format(self.id))
+                raise
```

Comment:
Wouldn't it be more informative if the error message was part printed with the `raise` statement?

Reply:
I'd like to match the existing behavior for this kind of failure in the Job class. Logging an error and raising the original exception also allows users to catch that exception and handle it as they wish. Either choice would be okay in general, but let's keep it this way for consistency. Thanks for the attention to detail!